src/rust/rust-proto/src/graplinc/grapl/api/event_source/v1beta1/client.rs

```diff
@@ -1,102 +1,83 @@
-    protobufs::graplinc::grapl::api::event_source::v1beta1::{
-        self as proto,
-        event_source_service_client::EventSourceServiceClient as EventSourceServiceClientProto,
-    },
-    protocol::{
-        endpoint::Endpoint,
-        error::GrpcClientError,
-        service_client::{
-            ConnectError,
+    graplinc::grapl::api::{
+        client::{
+            client_impl,
+            Client,
+            ClientError,
             Connectable,
         },
+        event_source::v1beta1 as native,
     },
+    protobufs::graplinc::grapl::api::event_source::v1beta1::event_source_service_client::EventSourceServiceClient,
 };
 
-pub type EventSourceServiceClientError = GrpcClientError;
+#[async_trait::async_trait]
+impl Connectable for EventSourceServiceClient<tonic::transport::Channel> {
+    async fn connect(endpoint: Endpoint) -> Result<Self, ClientError> {
+        Ok(Self::connect(endpoint).await?)
+    }
+}
 
 #[derive(Clone)]
-pub struct EventSourceServiceClient {
-    proto_client: EventSourceServiceClientProto<tonic::transport::Channel>,
-    executor: Executor,
+pub struct EventSourceClient {
+    client: Client<EventSourceServiceClient<tonic::transport::Channel>>,
 }
 
-#[async_trait::async_trait]
-impl Connectable for EventSourceServiceClient {
-    type Config = EventSourceClientConfig;
+impl client_impl::WithClient<EventSourceServiceClient<tonic::transport::Channel>>
+    for EventSourceClient
+{
     const SERVICE_NAME: &'static str = "graplinc.grapl.api.event_source.v1beta1.EventSourceService";
 
-    #[tracing::instrument(err)]
-    async fn connect_with_endpoint(endpoint: Endpoint) -> Result<Self, ConnectError> {
-        let executor = Executor::new(ExecutorConfig::new(Duration::from_secs(30)));
-        let proto_client = create_proto_client!(
-            executor,
-            EventSourceServiceClientProto<tonic::transport::Channel>,
-            endpoint,
-        );
-
-        Ok(EventSourceServiceClient {
-            executor,
-            proto_client,
-        })
+    fn with_client(client: Client<EventSourceServiceClient<tonic::transport::Channel>>) -> Self {
+        Self { client }
     }
 }
 
-impl EventSourceServiceClient {
+impl EventSourceClient {
     #[tracing::instrument(skip(self, request), err)]
     pub async fn create_event_source(
         &mut self,
         request: native::CreateEventSourceRequest,
-    ) -> Result<native::CreateEventSourceResponse, EventSourceServiceClientError> {
-        execute_client_rpc!(
-            self,
-            request,
-            create_event_source,
-            proto::CreateEventSourceRequest,
-            native::CreateEventSourceResponse,
-            RpcConfig::default(),
-        )
+    ) -> Result<native::CreateEventSourceResponse, ClientError> {
+        self.client
+            .execute(
+                request,
+                |status| status.code() == tonic::Code::Unavailable,
```

Comment:
Potential followup thing to do, not necessary:

it'd be nice to have somewhere an easily-accessible-but-still-needs-to-be-specified retry predicate like
```
fn retry_on_unavailable(status: tonic_or_rust_proto_i_forget::Status) {
  status.code() == tonic::Code::Unavailable
}
```
and you could just pass that function pointer;
but I'm not sure if an `fn` can be passed as an `Fn` (or maybe it needs to be casted?)

Reply:
PR checkpoint before lunch